File: re-ssim-components/re-ssim-caterpillar-application/src/main/java/com/shinonometn/re/ssim/caterpillar/application/ReSsimCaterpillarApplication.java
package com.shinonometn.re.ssim.caterpillar.application;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.core.task.TaskExecutor;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

import java.io.File;

@SpringBootApplication
public class ReSsimCaterpillarApplication {
    public static void main(String[] args) {
        SpringApplication.run(ReSsimCaterpillarApplication.class, args);
    }

    @Value("${application.data.path}")
    private String rootFolderPath;

    @Bean
    public TaskExecutor taskExecutor() {
        return new ThreadPoolTaskExecutor();
    }

    @Bean
    public File rootFolder() {
        File file = new File(rootFolderPath);
        if (!file.exists() && !file.mkdirs())
            throw new RuntimeException("Could not create folder " + file.getAbsolutePath() + ".");

        return file;
    }
}

File: re-ssim-components/re-ssim-caterpillar-application/src/main/java/com/shinonometn/re/ssim/caterpillar/application/ReSsimCaterpillarApplication.kt
package com.shinonometn.re.ssim.caterpillar.application

import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.core.task.TaskExecutor
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
import org.springframework.web.servlet.config.annotation.EnableWebMvc

import java.io.File

fun main(args: Array<String>) {
    SpringApplication.run(ReSsimCaterpillarApplication::class.java, *args)
}

@SpringBootApplication
@EnableWebMvc
open class ReSsimCaterpillarApplication {

    @Value("\${application.data.path}")
    private val rootFolderPath: String? = null

    @Bean
    open fun taskExecutor(): TaskExecutor {
        return ThreadPoolTaskExecutor()
    }

    @Bean
    open fun rootFolder(): File {
        val file = File(rootFolderPath!!)
        if (!file.exists() && !file.mkdirs())
            throw RuntimeException("Could not create folder " + file.absolutePath + ".")

        return file
    }
}
